glib-macros/src/object_subclass_attribute.rs

```diff
@@ -87,8 +87,36 @@ pub fn impl_object_subclass(input: &syn::ItemImpl) -> TokenStream {
             #class_opt
             #instance_opt
             #new_opt
-            #crate_ident::object_subclass_internal!();
             #(#items)*
         }
+
+        unsafe impl #crate_ident::subclass::types::ObjectSubclassType for #self_ty {
```

Comment:
The same change is also needed for `ObjectInterface` and the corresponding macro. Can you do that?

Otherwise looks like a good improvement to me, thanks :)